### ISSUE TITLE
Windows bat error handling broken and odd effects

### ISSUE THREAD

**Issue by hohwille** [devonfw/ide#1020]
### Expected behavior

When I run `setup.bat` and this fails, the setup should fail almost immediately after printing out the error message in red and I should *not* get a "green" message `Setup of devonfw-ide completed` at the end.

### Actual behavior

As you can see in 
![image](https://user-images.githubusercontent.com/1923119/207912509-e2de6e8a-3365-4c2f-bfc0-8b29b2d4d952.png)
or
![image](https://user-images.githubusercontent.com/1923119/207912710-b8ec7d13-671c-46dd-a9dc-b49198c2b087.png)
the error handling does not work.

Also notice that we get this line 5 times:
```
The operation completed successfully.
```
Where does that come from and how can we prevent this?

### Steps to reproduce (bug) / Use Case of feature request (enhancement)

1. Enforce some error (e.g. Open `regedit` and edit `Computer\HKEY_LOCAL_MACHINE\SOFTWARE\GitForWindows\InstallPath` by adding a `2` at the end) for force this error: https://github.com/devonfw/ide/pull/1019#issuecomment-1353333387
2. Run `setup.bat`

### Related/Dependent Issues

Only very slightly related is #1005

### Comments/Hints:
IMHO in `bat` scripts we do not actually produce and handle exit codes at all.

### Affected version:

* 2022.11.002

**Comment by hohwille** [comment]
For the record:
```
Failed to pip install pypiwin32 (pip)
...
Failed to run devon pip install azure-cli az
```
The first message is somewhat confusing. When we call `doRunCommand` you can optionally provide a text that describes the called command in a more human understandable way. IMHO we are passing `pip install pypiwin32` here what is not a really good idea.

**Comment by hohwille** [comment]
I can easily fix the error handling.
However, the printing of `The operation completed successfully.` seems to come from every successful call of CMD:
1. setup.bat
2. setup.bat thinking it runs in powershell running itself again in CMD
3. devon.bat ide setup
4. bash called to execute `devon ide setup`.
5. `devon` alias calling `ide` commandlet.

If this really matches the 5 times of the success message and I am counting and thinking right, then also the bash process seems to trigger this message at the end so more or less I get the message for every child-process that succeeds.

I found people having the same problem and asking how to suppress this but incredibly I found no answers yet:
https://msfn.org/board/topic/100693-suppress-operation-completed-successfully-in-batch-files/
Well, there are some answers but those are only hacks like redirecting the output to NUL or a temporary file.
Also the CMD usage does not say anything about this issue:
https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/cmd

**Comment by hohwille** [comment]
OK, I will have to create a new story for this strange outputs like `The operation completed successfully.` or `ERROR: Error opening the file. There may be a disk or file system error.`.
Further #866 is not properly fixed. I have meanwhile fixed #1018 but the entire approach to detect if we are inside powershell is not valid or working reliable. Simply start a new `CMD` in windows terminal and execute this command:
```
echo %PSMODULEPATH%
```
You will see that it contains what we expect if we are called from within powershell but IMHO where are not, if we simply open a new CMD tab in Windows Terminal.
